### PR TITLE
Add template type to Suspension

### PR DESCRIPTION
**Pull request by trowski**

### CODE REVIEW

File: src/EventLoop.php [hunk 1]
```diff
@@ -363,7 +362,9 @@ public static function getDriver(): Driver
      *
      * Calls from the same fiber will return the same suspension object.
      *
-     * @return Suspension
+     * @template T
+     *
+     * @return Suspension<T>
```
Comment (kelunik) [inline src/EventLoop.php]: How should `T` be inferred here? It's just `Suspension<mixed>`, no?

Comment (trowski) [reply src/EventLoop.php]: Unfortunately it can't be inferred. Should this be annotated as `Suspension<mixed>` instead?

Comment (kelunik) [reply src/EventLoop.php]: I guess this will happen anyway inside static analysis, no?

Comment (trowski) [reply src/EventLoop.php]: Probably, yeah, so this needn't change then.

**Review by kelunik — APPROVED** [review]

**Pull request merged**